qemu/tests/virtio_console.py

```diff
@@ -632,10 +645,17 @@ def _port_replug(device, port_idx):
             # when replugging port from pci to different pci. We should
             # either use symlinks (as in Windows) or replug with the busname
             port = ports[port_idx]
-            vm.monitor.cmd('device_del %s' % port.qemu_id)
+            qdev = vm.devices
+            portdev = qdev.get_by_params({"name": port.qemu_id})
+            if not portdev:
+                test.error("No port named %s" % port.qemu_id)
+            chardev = portdev[0].get_param("chardev")
+            vm.monitor.cmd('device_del %s' % get_portid(vm, port.qemu_id))
             time.sleep(intr_time)
-            vm.monitor.cmd('device_add %s,id=%s,chardev=dev%s,name=%s'
-                           % (device, port.qemu_id, port.qemu_id, port.name))
+            if not chardev:
+                test.error("No chardev in guest for port %s" % port.qemu_id)
+            vm.monitor.cmd('device_add %s,id=%s,chardev=%s,name=%s'
+                           % (device, get_portid(vm, port.qemu_id), chardev, port.name))
```

Comment:
You get portid by the function many times, use a variant record it and just call the variant will be more efficiency.

Reply:
Thanks, I will update it.